### DISCUSSION TITLE
Further Development Process

### DISCUSSION THREAD

**Discussion by AndyPro720** [Unified-Development-Environment/Unified_HAL#3] [Ideas]
We'll be keeping the tasks for our reference on notion, this place to discuss what is needed. We will need to get on a call soon to nail the basics.

@rish292

**Comment by AndyPro720** [comment]
So for starters, 

* Each branch/vendor library format needs to documented. (can be done as rough on notion, tested and then added to respective branch)
* Makefile working and implementation, probably linked with the first point.
* And finally, link relevant resources on header file and #elseif syntax. Work it properly in our existing system. 
*

**Reply by rish292** [reply]
Hey @AndyPro720 . Yes, you had mentioned this but you'll have to explain it more in depth here. The live explanation you gave a few days back gave me a rough idea but I won't get any basics cleared on a call/on the spot as I have discussed in the past with you so let's continue the discussion here. What do you mean by "document a library format as rough on notion" ? If you've tried something and think it could be worked upon then let me in on it via Github. In order to effectively understand and retain what you are trying out exactly I'll have to intuitively go through it at my own end using whatever demo you have worked out. 

Also, I am looking at the "addition of specs to makefile" aspect for the source and header files and I'll update if I have some leads.

**Reply by AndyPro720** [reply]
> What do you mean by "document a library format as rough on notion" ?

@rish292 Here, I meant a documented the structure of each dev board importable library(structure) before we actually create it on the repository. 

Also, we'll need to keep a separate branch/folder for the documentation of each function/lib, so that can be easily recreated across each of the different platforms without errors. We could use the WIKI section in github(have put up a sample) or simply create a notion page. 

 Let me know what you'd suggest.

**Reply by rish292** [reply]
Yea ig you can go ahead with the GitHub Wiki. Although there are organisations which also have a separate repo for the documentation so that could be done as well. @AndyPro720

**Reply by rish292** [reply]
I also think the structure would fall in place as the code on the repo progresses rather having one prior.

**Comment by AndyPro720** [comment]
Links, the basic, not exhaustive 
 
[default Parameters in C](https://stackoverflow.com/questions/1472138/c-default-arguments)
[idea of makefile](https://stackoverflow.com/questions/5559250/c-error-undefined-reference-to-function-but-it-is-defined)
[libs on Arduino](https://docs.arduino.cc/software/ide-v1/tutorials/installing-libraries)

**Reply by AndyPro720** [reply]
[How to write Libs](https://docs.arduino.cc/learn/contributions/arduino-creating-library-guide) Great read
[API/code styling](https://docs.arduino.cc/learn/contributions/arduino-library-style-guide)
@rish292

**Reply by AndyPro720** [reply]
[refer to class and namespaces section](https://www.notion.so/scientia72/Coding-Paradigm-C-Cpp-af190394bd7c475980943d130f760815#6e26fe1fafbb462781e4f330259b906d)

**Reply by rish292** [reply]
https://github.com/sudar/Arduino-Makefile

**Comment by rish292** [comment]
![Screenshot 2023-05-01 at 9 16 04 PM](https://user-images.githubusercontent.com/104636130/235481168-d673fe07-b283-47da-a15b-d9687ecbe01a.png)
I think the strike out might indicate that the unified_hal.h file is able to appropriately identify which hal file to use based on the label.
This is from the test folder which I created and I haven't made any changes.

**Reply by rish292** [reply]
![Screenshot 2023-05-01 at 9 27 01 PM](https://user-images.githubusercontent.com/104636130/235483072-963c5b40-8b3b-49a7-a1fe-021b6d3832de.png)
So these are the object files for the respective .c files

**Reply by rish292** [reply]
![Screenshot 2023-05-01 at 9 28 49 PM](https://user-images.githubusercontent.com/104636130/235483371-86c29553-a6b7-479c-b82b-73bbb163157b.png)
I couldn't find any equivalent for the source files but instead of the list of files i found the subdir of all source files in sources.mk

**Reply by rish292** [reply]
![Screenshot 2023-05-01 at 9 33 35 PM](https://user-images.githubusercontent.com/104636130/235484068-df611498-6159-4238-9562-55d4d22d2949.png)
I couldn't see where the inc/include files are listed....this is from the makefile and the source ones are included

**Reply by rish292** [reply]
You can find all of these in the Debug Folder @AndyPro720
